Add ability to use different converters via factory (#359)
tarantool/cartridge-java: CHANGELOG.md

## [Unreleased]
- Close public access to TarantoolResult*Impl ([#326](https://github.com/tarantool/cartridge-java/issues/326))
- Add deep copy instead of shallow copy in default message pack mapper

> - Add factory builder to be able making mapper hierarchy

Reviewer: ```suggestion
- Add a factory builder for constructing mapper hierarchies
```
